### ISSUE TITLE
amplify mock not setting API_*_GRAPHQLAPIENDPOINTOUTPUT env var correctly in lambda resolver

### ISSUE THREAD

**Issue by hisham** [aws-amplify/amplify-cli#6548]
**Describe the bug**
I have a graphql lambda resolver that as part of its processing does a create mutation via the same graphql API. 

I am currently testing this lambda resolver via amplify mock. Everything works until the lambda attempts to do a graphql mutation. 

The error I get is the following: 

```
Lambda Data source failed with the following error

{
  type: 'Lambda:Unhandled',
  message: {
    message: 'Request failed with status code 403',
    name: 'Error',
    stack: 'Error: Request failed with status code 403\n' +
      '    at createError (/Users/hisham/src/ess-app/amplify/backend/function/EssGraphQLResolverFn/src/node_modules/axios/lib/core/createError.js:16:15)\n' +
      '    at settle (/Users/hisham/src/ess-app/amplify/backend/function/EssGraphQLResolverFn/src/node_modules/axios/lib/core/settle.js:17:12)\n' +
      '    at IncomingMessage.handleStreamEnd (/Users/hisham/src/ess-app/amplify/backend/function/EssGraphQLResolverFn/src/node_modules/axios/lib/adapters/http.js:260:11)\n' +
      '    at IncomingMessage.emit (events.js:327:22)\n' +
      '    at IncomingMessage.EventEmitter.emit (domain.js:467:12)\n' +
      '    at endReadableNT (internal/streams/readable.js:1327:12)\n' +
      '    at processTicksAndRejections (internal/process/task_queues.js:80:21)',
    config: {
      url: 'https://<myapi>.appsync-api.us-east-1.amazonaws.com/graphql',
      method: 'post',
      data: '{"query":"mutation....  
```

Looking at the url above, it is supposed to be http://192.168.86.25:20002 since I am running amplify mock, but instead it's my actual non-mock aws appsync endpoint. It does not appear that amplify mock sets the API...GRAPHQLAPIENDPOINTOUTPUT env var correctly.

**Amplify CLI Version**
4.41.2

**To Reproduce**
Create a lambda resolver that in turn calls the same graphql api with a mutation or query, and test via amplify mock.

**Comment by kaustavghosh06** [comment]
@hisham So you have a Lambda resolver as a part of your schema which is calling the GraphQL endpoint as well? 
Currently we don't set any environment variables locally for you - but you can set it locally as a process environment for local mocking purposes. We're working on a larger project for setting environment variables locally for lambda functions - cc @edwardfoyle

**Comment by hisham** [comment]
Yes that's correct. Ok I'll try your suggestion next time I use amplify mock.

**Comment by hisham** [comment]
@kaustavghosh06 running the following does not work: 

`API_<myapi>_GRAPHQLAPIENDPOINTOUTPUT=http://localhost:20002 amplify mock`

It seems the amplify mock code overrides it with the incorrect url. 

I would have to set this env var in the lambda code instead, and then remove it when I do amplify push. Is there a way within lambda to know if it's in a mock environment? I see you set ENV to NONE and REGION to us-east-1-fake, and I can check other things in the env, but I wonder if there's an official way.

**Comment by kaustavghosh06** [comment]
Got it. This would PR would potentially fix your issue @hisham - https://github.com/aws-amplify/amplify-cli/pull/6551

**Comment by edwardfoyle** [comment]
A fix for this has been released in version 4.44.1 of the CLI. Please upgrade to the latest version to try it out.
